Remove etcd SRV lookup from bootkube.sh

cybertron opened the pull request:
Upstream didn't like having this in the bootstrap script, and since
alternative solutions would be tricky to implement in time for 4.2,
we chose to remove it and add a requirement that the master nodes
be named in a predictable manner, specifically master-{0,1,2}. This
results in our etcd records being named the same as the other
openshift platforms and allows us to use the same logic for specifying
the etcd members as they do.

cybertron committed:
Remove etcd SRV lookup from bootkube.sh

Upstream didn't like having this in the bootstrap script, and since
alternative solutions would be tricky to implement in time for 4.2,
we chose to remove it and add a requirement that the master nodes
be named in a predictable manner, specifically master-{0,1,2}. This
results in our etcd records being named the same as the other
openshift platforms and allows us to use the same logic for specifying
the etcd members as they do.

stbenjam commented:
Does this mean we could move the MCO block to the same location as in openshift/installer? Currentlyours runs much earlier (and causes a lot of rebase issues).

metal3ci commented:
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/819/

cybertron commented:
Probably, but I haven't tried it. I'll test it locally and add it to this PR if it works.

cybertron committed:
Move MCO bootstrap back to its original spot in bootkube

We had originally moved MCO earlier in the bootstrap process because
we needed it to stand up the mdns services on the masters so we could
retrieve their etcd DNS names. Now that we're not doing that lookup
during bootstrap we can use the stock MCO bootstrap order again.

cybertron commented:
Moving the MCO bootstrap back seems to have worked fine, so I've added that to this PR.

stbenjam reviewed (approved):
Thank you! LGTM if CI passes.

metal3ci commented:
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/821/

metal3ci commented:
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/822/

hardys commented:
Kicked off CI again ref http://10.8.144.11:8080/job/dev-tools/824/ as I think the CI failure was due to the unit tests, fixed via https://github.com/openshift-metal3/kni-installer/pull/131

metal3ci commented:
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/824/

stbenjam commented:
Awesome, thanks!
